Crash when project uses Glide v4.11.0 and stream-chat-android v4.3.2

Encountered a crash when running a project that depends on glide v4.11.0 and stream-chat-android v4.3.2, running the project will crash as soon as I try to load an image with glide. 

If I downgrade the app's glide version in the build.gradle to v4.9.0 then everything works fine or if I remove the stream-chat-android dependency and keep glide at v4.11.0 then that also works.

Any way around this issue? I would like to use glide v4.11.0 without stream-chat-android causing crashes.

Here is a sample project that reproduces the crash, the app should crash on startup every time with the below stack trace.
https://github.com/VGJohn/StreamGlideCrash

Stack trace:

`2020-05-11 20:02:28.329 1960-1960/com.test.streamglidetest E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.test.streamglidetest, PID: 1960
    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.test.streamglidetest/com.test.streamglidetest.MainActivity}: java.lang.IllegalStateException: GeneratedAppGlideModuleImpl is implemented incorrectly. If you've manually implemented this class, remove your implementation. The Annotation processor will generate a correct implementation.
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3108)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3251)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1948)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7045)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:964)
     Caused by: java.lang.IllegalStateException: GeneratedAppGlideModuleImpl is implemented incorrectly. If you've manually implemented this class, remove your implementation. The Annotation processor will generate a correct implementation.
        at com.bumptech.glide.Glide.throwIncorrectGlideModule(Glide.java:344)
        at com.bumptech.glide.Glide.getAnnotationGeneratedGlideModules(Glide.java:336)
        at com.bumptech.glide.Glide.get(Glide.java:179)
        at com.bumptech.glide.Glide.getRetriever(Glide.java:749)
        at com.bumptech.glide.Glide.with(Glide.java:801)
        at com.test.streamglidetest.MainActivity.onCreate(MainActivity.kt:15)
        at android.app.Activity.performCreate(Activity.java:7327)
        at android.app.Activity.performCreate(Activity.java:7318)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1271)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3088)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3251) 
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1948) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:214) 
        at android.app.ActivityThread.main(ActivityThread.java:7045) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:964) 
     Caused by: java.lang.NoSuchMethodException: com.bumptech.glide.GeneratedAppGlideModuleImpl.<init> [class android.content.Context]
        at java.lang.Class.getConstructor0(Class.java:2328)
        at java.lang.Class.getDeclaredConstructor(Class.java:2167)
        at com.bumptech.glide.Glide.getAnnotationGeneratedGlideModules(Glide.java:320)`

--- a/build.gradle
+++ b/build.gradle
@@ -9,7 +9,7 @@ buildscript {
         appCompatVersion = '1.1.0'
         constraintLayoutVersion = '2.0.0-beta2'
         espressoVersion = '3.2.0'
-        glideVersion = '4.9.0'
+        glideVersion = '4.11.0'
         googleServiceVersion = '4.3.2'
         androidGradlePluginVersion = '3.5.3'
         gradlePluginVersion = '2.1'

--- a/library/build.gradle
+++ b/library/build.gradle
@@ -84,10 +84,10 @@ dependencies {
     implementation 'com.squareup.retrofit2:converter-gson:2.8.1'
 
     implementation 'com.google.code.gson:gson:2.8.6'
-    implementation 'com.aminography:redirectglide:1.0.3'
+    implementation 'com.aminography:redirectglide:2.0.1'
 
-    implementation 'com.github.bumptech.glide:glide:4.9.0'
-    kapt 'com.github.bumptech.glide:compiler:4.9.0'
+    implementation "com.github.bumptech.glide:glide:$rootProject.glideVersion"
+    kapt "com.github.bumptech.glide:compiler:$rootProject.glideVersion"
 
     implementation 'com.github.duanhong169:drawabletoolbox:1.0.7'
     implementation "io.noties.markwon:core:$markdown_version"

